tests/jvm/src/test/scala/cats/effect/IOPlatformSpecification.scala

```diff
@@ -724,6 +725,48 @@ trait IOPlatformSpecification extends DetectPlatform { self: BaseSpec with Scala
         poller.wasInterrupted.get() must beTrue
       }
 
+      "handle mixed-mode poller/simple interruption with complex timers" in {
+        val delegate = unsafe.SelectorSystem()
```

Comment:
This will merge into 3.x better.

```suggestion
        val delegate = IORuntime.createDefaultPollingSystem()
```